frontend/src/services/Api.ts

```diff
@@ -494,24 +534,26 @@ export const getNamespaceServiceHealth = (
   );
 };
 
-export const getNamespaceWorkloadHealth = (
+export const getNamespaceWorkloadHealth = async (
```

Comment:
I guess adding `async` here is mainly declarative?  I suppose we could add `async` to almost all of the functions in here, as even the ones that don't have explicit Promise return values are still returning promises (via `return newRequest`)

Reply:
The code editor suggested to add async when the method returns `Promise<XX>`but not when the method returns `Promise<Response<XX>>`. I don't know why. You are right, this async should be done in all methods since they are returning promises. I also do not like that we use our custom `Response` type in some methods and `AxiosResponse` in others. The same happens with `AxiosError`, we should have our own custom `Error` type like we have with `Response`. Doing all this stuff is out of scope of this PR (I have refactored enough I think), so I will create an issue to handle this and finish the refactorization of Kiali API in the frontend.

Reply:
I agree.  I think I tried doing something like this suggested refactor at one point, and it turned into a can or worms. The resulting type checking issues became a little overwhelming, so I abandoned the effort.  Perhaps with all of our good refactoring and more strict typing, it would work better now, but be prepared for a headache :-)